### PR TITLE
New fake trajectory system

### LINKED ISSUE #15: Re-design fake systems to not rely on Sims
At the moment we require Sims to provide us with files containing sensor readings to test our software. This is problematic because once we change the pod's behaviour, e.g. by adding a state to the state machine, the sims data is off and we cannot simulate a run. This could be fixed by defining a ground truth displacement/velocity/acceleration curve that can then be used to generate sensor readings. This is close to what Sims are doing anyways so we can just rewrite their MATLAB code into a C++ component.

This could even incorporate motor controls as we could make the acceleration dependent on the motors and what they are currently doing.

In any case, Software should not rely on one particular pod design.

### PR DESCRIPTION
**Pull request by MiltFra**
Aims to centralise the fake value generation for the trajectory sensors (IMUs, Keyence, wheel encoders) by having a central (i.e. "ground truth") instance for all the relevant data.

This also changes the clang-format include grouping regex to allow underscores in more groups (e.g. `#include <unordered_map>`).

Fixes #15, #16.

# TODO

- [x] Test keyence config parsing
- [x] Add keyence failures
- [x] Test keyence failures
- [x] Add wheel encoder config parsing
- [x] Test wheel encoder config parsing
- [x] Add wheel encoder failures
- [x] Test wheel encoder failures
- [x] Add IMU config parsing
- [x] Test IMU config parsing
- [x] Add IMU failures
- [x] Test IMU failures
- [x] Remove system command line flags for fake system
- [x] Add JSON system config

### CODE REVIEW

File: src/data/data.hpp [hunk 1]
```diff
@@ -37,52 +30,49 @@ struct Module {
 // -------------------------------------------------------------------------------------------------
 // Navigation
 // -------------------------------------------------------------------------------------------------
-typedef float nav_t;
-typedef Vector<nav_t, 3> NavigationVector;
+using nav_t            = float;
+using NavigationVector = utils::math::Vector<nav_t, 3>;
 struct Navigation : public Module {
   static constexpr nav_t kMaximumVelocity = 100;     // m/s
```
Comment (maxguy2001) [inline src/data/data.hpp:36]: Some constants are quite outdated here (max velocity, run length, braking distance(?)).

File: src/data/data.hpp [hunk 2]
```diff
@@ -37,52 +30,49 @@ struct Module {
...
   static constexpr nav_t kRunLength       = 1250.0;  // m
   static constexpr nav_t kBrakingBuffer   = 20.0;    // m
-  nav_t displacement;                                // m
-  nav_t velocity;                                    // m/s
-  nav_t acceleration;                                // m/s^2
-  nav_t emergency_braking_distance;                  // m
-  nav_t braking_distance = 750;                      // m
+  // TODO(miltfra): Fix wheel and keyence distances
+  static constexpr nav_t kWheelCircumfrence = 3.14 * 0.1;  // m
+  static constexpr nav_t kStripeDistance    = 50;          // m
```
Comment (maxguy2001) [inline src/data/data.hpp:41]: kStripeDistance also defined in navigation.hpp with conflicting values?

File: src/data/data.hpp [hunk 3]
```diff
@@ -37,52 +30,49 @@ struct Module {
...
   static constexpr nav_t kMaximumVelocity = 100;     // m/s
   static constexpr nav_t kRunLength       = 1250.0;  // m
   static constexpr nav_t kBrakingBuffer   = 20.0;    // m
-  nav_t displacement;                                // m
-  nav_t velocity;                                    // m/s
-  nav_t acceleration;                                // m/s^2
-  nav_t emergency_braking_distance;                  // m
-  nav_t braking_distance = 750;                      // m
+  // TODO(miltfra): Fix wheel and keyence distances
+  static constexpr nav_t kWheelCircumfrence = 3.14 * 0.1;  // m
```
Comment (maxguy2001) [inline src/data/data.hpp:40]: More decimals on pi? Error on same order of magnitude as some of our variances/uncertainties.

File: src/data/data.hpp (hunk 1 above)
Comment (MiltFra) [reply src/data/data.hpp]: To be honest, these are mainly used by state machine. In the future, we may want to move these into a config file but for now, they're fine here.

**Review by kshxtij — APPROVED** [review]

**Review by maxguy2001 — APPROVED** [review]

**Pull request merged**